numbat/src/vm.rs

```diff
@@ -615,6 +652,59 @@ impl Vm {
                     };
                     self.push_quantity(result.map_err(RuntimeError::QuantityError)?);
                 }
+                op @ (Op::AddDateTime | Op::SubDateTime) => {
+                    let rhs = self.pop_quantity();
+                    let lhs = self.pop_datetime();
+
+                    // for time, the base unit is in seconds
+                    let base = rhs.to_base_unit_representation();
+                    let seconds_f = base.unsafe_value().to_f64();
+
+                    // TODO check that this handles negative durations correctly
+                    let duration = chrono::Duration::seconds(seconds_f.trunc() as i64)
+                        + chrono::Duration::nanoseconds(
+                            (seconds_f.fract() * 1_000_000_000f64).round() as i64,
+                        );
+
+                    self.push(Value::DateTime(
+                        match op {
+                            Op::AddDateTime => lhs + duration,
+                            Op::SubDateTime => lhs - duration,
+                            _ => unreachable!(),
+                        },
+                        chrono::Local::now().offset().fix(),
+                    ));
+                }
+                Op::DiffDateTime => {
+                    let unit = self.pop_quantity();
+                    let rhs = self.pop_datetime();
+                    let lhs = self.pop_datetime();
+
+                    let duration = lhs - rhs;
+                    // TODO error handling
+                    let duration = duration.num_nanoseconds().unwrap() as f64 / 1_000_000_000f64;
+
+                    let ret = Value::Quantity(Quantity::new(
+                        Number::from_f64(duration),
+                        unit.unit().clone(),
+                    ));
+
+                    self.push(ret);
+                }
+                Op::ConvertDateTime => {
+                    let rhs = self.pop_string();
+                    let lhs = self.pop_datetime();
+
+                    // TODO how to handle errors, and is this `chrono_tz` crate the best choice?
```

Comment:
We should at least make a wrong timezone a runtime error here, if not possible earlier. Right now, if you do `now() -> "Florida"`, it will fall back to UTC silently.